lib/src/cardano_version.dart

```diff
@@ -0,0 +1,34 @@
+import 'package:flutter/foundation.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'cardano_version.freezed.dart';
+
+@freezed
+class CardanoVersion with _$CardanoVersion {
+  const CardanoVersion._();
+
+  const factory CardanoVersion({
+    required bool testMode,
+    required int versionMajor,
+    required int versionMinor,
+    required int versionPatch,
+    required bool locked,
+  }) = _CardanoVersion;
+
+  /// Get the version code.
+  int get versionCode =>
+      versionMajor * 10000 + versionMinor * 100 + versionPatch;
+
+  /// Get the version name.
+  String get versionName => '$versionMajor.$versionMinor.$versionPatch';
+
+  static CardanoVersion fromVersionCode(int versionCode) {
```

Comment:
Not a big deal but such "custom constructors" are usually defined as factories in Dart. Also it could be done as a lamdba expression (using `=>`).

I believe the code should change to
```dart
const factory CardanoVersion.fromVersionCode(int versionCode) => CardanoVersion(
      testMode: false,
      versionMajor: versionCode ~/ 10000,
      versionMinor: (versionCode % 10000) ~/ 100,
      versionPatch: versionCode % 100,
      locked: false,
    );
```

P.s. If it complains about using `const`, just remove it.

Reply:
done